scripts/unicode.py

```diff
@@ -263,7 +305,26 @@ def emit_enums(f, script_list, extension_list, longforms):
         f.write("            Script::%s => \"%s\",\n" % (longforms[script], script))
     f.write("""        }
     }
+""")
 
+    # Generate impl of `inner_from_short_name`
+    f.write("""
+    #[inline]
+    pub(crate) fn inner_from_short_name(input: &str) -> Option<Self> {
+        match input {
+            "" => Some(Script::Unknown),
```

Comment:
should this branch exist? Perhaps we should not return an Option and use Unknown as the catch all

Reply:
I think `Option` may be helpful, because if some folk will accidentally use `from_short_name` instead of `from_full_name`they'll get a `None`, which is easier to find early.
Like, my motivation was to held both `short_name . from_short_name . unwrap = id` and `from_short_name . unwrap . short_name = id`.

Reply:
However, I'm not insisting, if you think that it's better to not use `Option` here, I'll be happy to change the code.

Reply:
I'm okay returning an Option but we shouldn't have the empty branch imo